src/pages/docs/getting-started/java.mdx

```diff
@@ -0,0 +1,274 @@
+```
+</Code>
+
+## Step 3: Join the presence set <a id="step-3"/>
+
+Presence enables clients to be aware of one another if they are present on the same channel. You can then show clients who else is online, provide a custom status update for each, and notify the channel when someone goes offline.
+
+Add the following lines to your `main` method, above the line `// Keep the program running`, to subscribe to, and join, the presence set of the channel. Then run it with `./gradlew run`:
+
+<Code>
+```java
+// Subscribe to presence events on the channel
+channel.presence.subscribe(presenceMessage -> {
+    System.out.println("Event type: " + presenceMessage.action.name() +
+                      " from " + presenceMessage.clientId +
+                      " with the data " + presenceMessage.data);
+});
+
+// Enter the presence set
+channel.presence.enter("I'm here!", null);
+```
+</Code>
+
+You can have another client join the presence set using the Ably CLI:
+
+<Code>
+```shell
+ably channels presence enter my-first-channel --client-id "my-cli" --data '{"status":"learning about Ably!"}'
+```
+</Code>
+
+## Step 4: Retrieve message history <a id="step-4"/>
+
+You can retrieve previously sent messages using the history feature. Ably stores all messages for 2 minutes by default in the event a client experiences network connectivity issues. This can be extended for longer if required.
+
+If more than 2 minutes has passed since you published a regular message (excluding the presence events), then publish some more before trying out history. You can use the Pub/Sub SDK, Ably CLI or the dev console to do this.
+
+For example, using the Ably CLI to publish 5 messages:
+
+<Code>
+```shell
+ably channels publish --count 5 my-first-channel "Message number {{.Count}}" --name "myEvent"
+```
+</Code>
+
+Add the following lines to your `main` method, above the line `// Keep the program running`, to retrieve any messages that were recently published to the channel. Then run it with `./gradlew run`:
+
+<Code>
+```java
+// Retrieve message history
+PaginatedResult<Message> resultPage = channel.history(null);
+Message[] messages = resultPage.items();
+if (messages.length > 0) {
+    System.out.println("Message History:");
+    for (Message message : messages) {
+        System.out.println(message.data);
+    }
+} else {
+    System.out.println("No messages in history.");
+}
+```
+</Code>
+
+The output will look similar to the following:
+
+<Code>
+```json
+[
+  'Message number 5',
+  'Message number 4',
+  'Message number 3',
+  'Message number 2',
+  'Message number 1'
+]
+```
+</Code>
+
+## Step 5: Close the connection <a id="step-5"/>
+
+Connections are automatically closed approximately 2 minutes after no heartbeat is detected by Ably. Explicitly closing connections when they are no longer needed is good practice to help save costs. It will also remove all listeners that were registered by the client.
```

Comment:
> It will also remove all listeners that were registered by the client

sorry missed it, it's not true, we don't remove listeners when connection is closed. You can resurrect client by calling `connect()` so `close()` is actually works as `disconnect()`. Can you check this for other get-started guides as well?